src/DGtal/geometry/curves/estimation/CLMSTTangentFromDSS.h

```diff
@@ -0,0 +1,95 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file CLMSTTangentFromDSS.h
+ * @author Kacper Pluta (\c kacper.pluta@esiee.fr )
+ * Laboratoire d'Informatique Gaspard-Monge - LIGM, France
+ *
+ * @date 2015/09/12
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(CLMSTTangentFromDSS_RECURSES)
+#error Recursive header files inclusion detected in CLMSTTangentFromDSS.h
+#else // defined(CLMSTTangentFromDSS_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define CLMSTTangentFromDSS_RECURSES
+
+#if !defined CLMSTTangentFromDSS_h
+/** Prevents repeated inclusion of headers. */
+#define CLMSTTangentFromDSS_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include "DGtal/base/Common.h"
+#include "boost/concept_check.hpp"
+#include "DGtal/base/ConceptUtils.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+namespace concepts
+{
+  
+  /////////////////////////////////////////////////////////////////////////////
+  // class CLMSTTangentFromDSS
+  /**
+   * Description of \b concept '\b CLMSTTangentFromDSS' <p>
+   * @ingroup Concepts
+   * @brief Aim: Defines the concept describing a functor which calculate a direction
+   * of the 2D DSS and an eccentricity of a given point in this DSS.
+   * 
+   * @tparam T the type that should be a model of CLMSTTangentFromDSS.
+   */
```

Comment:
Please use Concept models, a lot of information is missing (usage, requirements...)